File: Iteration 3/Grade.py
from Course import Course


class Grade:
    def __init__(self, course : Course, grade_value):
        self.__course = course 
        self.__grade_value = grade_value 

    # Getter methods
    def get_course(self):
        return self.__course

    def get_grade_value(self):
        return self.__grade_value

    # toString method (or __str__ in Python) to provide a string representation of the grade
    def __str__(self):
        # Check if the course is a valid Course object
        course_info : Course = self.__course
        grade_info = self.__grade_value if self.__grade_value else "No grade assigned"

        if course_info:
            # Ensure that the course_info is a Course object, then get course_name
            return f"Grade - Course: {course_info.get_course_name()}, Grade: {grade_info}"
        else:
            return f"Grade - Course: No course assigned, Grade: {grade_info}"

File: Iteration 1/CSE3063F24P1_GRP14/src/Grade.java
import com.fasterxml.jackson.annotation.JsonProperty;

public class Grade {
    @JsonProperty("course")
    private Course course;

    @JsonProperty("gradeValue")
    private String gradeValue;

    public Grade(Course course, String gradeValue) {
        this.course = course;
        this.gradeValue = gradeValue;
    }
    public Grade() {}
    // Getters and setters
    public Course getCourse() {
        return course;
    }

    public String getGradeValue() {
        return gradeValue;
    }


    @Override
    public String toString() {
        return String.format("%-10s %-30s %-10s\n", course.getCourseId(), course.getCourseName(),gradeValue);
    }


}